### ISSUE TITLE
Review use of `write_table` in registration tasks

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#594]
As of https://github.com/fractal-analytics-platform/fractal-tasks-core/pull/581 and v0.13.1, we use `write_table` in `calculate_registration_image_based.` and `apply_registration_to_ROI_tables` tasks.
I'm confused by the fact that we call it as in
```python
# calculate_registration_image_based.py
    write_table(
        image_group,
        roi_table,
        new_ROI_table,
        overwrite=True,
        table_attrs=dict(type="ngff:region_table"),
    )

# apply_registration_to_ROI_tables.py
        image_group = zarr.group(f"{well_zarr}/{acq}")
        write_table(
            image_group,
            new_roi_table,
            shifted_rois[acq],
            table_attrs=dict(type="ngff:region_table"),
        )
```

This seems to contradict the following check in `write_table`:
```python

    # Optionally update attributes of the new-table zarr group
    if table_attrs is not None:
        if table_attrs.get("type") == "ngff:region_table":
            # Verify whether we comply with a proposed change to the OME-NGFF
            # table specs (https://github.com/ome/ngff/pull/64)
            try:
                table_attrs["instance_key"]
                table_attrs["region"]["path"]
            except KeyError as e:
                logger.warning(
                    f"The table_attrs parameter of write_elem has "
                    "type='ngff:region_table' but does not comply with the "
                    "proposed table specs. "
                    f"Original error: KeyError: {str(e)}"
                )
        # Update table_group attributes with table_attrs key/value pairs
        table_group.attrs.update(**table_attrs)
```

Question: are the registration tasks actually working with v0.13.1? Do we get the appropriate warning in the logs?

**Comment by jluethi** [comment]
I'll test this! If it doesn't work anymore, we'll want to improve our testing of those tasks :)

**Comment by tcompa** [comment]
> I'll test this! If it doesn't work anymore, we'll want to improve our testing of those tasks :)

Thanks. My expectation is that everything runs fine, but we do observe those KeyError warnings.

**Comment by jluethi** [comment]
I can now confirm. Example 3 still runs, but we get the following warning in the logs:
```
2023-10-27 16:44:41,936; WARNING; The table_attrs parameter of write_elem has type='ngff:region_table' but does not comply with the proposed table specs. Original error: KeyError: 'instance_key'
```

It's in all 3 registration tasks:
Calculate registration image-based
Apply Registration to ROI table
Apply Registration to Images

**Comment by tcompa** [comment]
Thanks for checking!

We'll come back to this issue after we finalize the definition of the V1 specs (see e.g. https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/529#issuecomment-1782621804).

**Comment by tcompa** [comment]
In `calculate_registration_image_based`, we should check that the reference ROI table has the appropriate type. According to V1 specs, this should be either `"roi_table"` or `"masking_roi_table"`, but we should also accept `None` or `"ngff:region_table"` for backwards compatibility.

**Comment by tcompa** [comment]
> In `calculate_registration_image_based`, we should check that the reference ROI table has the appropriate type. According to V1 specs, this should be either `"roi_table"` or `"masking_roi_table"`, but we should also accept `None` or `"ngff:region_table"` for backwards compatibility.

Ref https://github.com/fractal-analytics-platform/fractal-tasks-core/commit/c49feadb7580fee3349dfec3d74c98d37a04871d

**Comment by tcompa** [comment]
For the record, `apply_registration_to_image.py` already propagates old attrs:
```python
        for table in table_dict.keys():
            logger.info(f"Copying table: {table}")
            # Get the relevant metadata of the Zarr table & add it
            # See issue #516 for the need for this workaround
            try:
                old_table_group = zarr.open_group(table_dict[table], mode="r")
            except zarr.errors.GroupNotFoundError:
                time.sleep(5)
                old_table_group = zarr.open_group(table_dict[table], mode="r")
            # Write the Zarr table
            curr_table = ad.read_zarr(table_dict[table])
            write_table(
                new_image_group,
                table,
                curr_table,
                table_attrs=old_table_group.attrs.asdict(),
            )
```

**Comment by tcompa** [comment]
https://github.com/fractal-analytics-platform/fractal-tasks-core/commit/3fe29a116042ca1c2e3d1edb5e7d5436dd9d93d5 takes care of `apply_registration_to_ROI_tables`.

Missing: re-add the same check as in https://github.com/fractal-analytics-platform/fractal-tasks-core/commit/c49feadb7580fee3349dfec3d74c98d37a04871d (that is: factor it out into another module).

**Comment by tcompa** [comment]
> [3fe29a1](https://github.com/fractal-analytics-platform/fractal-tasks-core/commit/3fe29a116042ca1c2e3d1edb5e7d5436dd9d93d5) takes care of `apply_registration_to_ROI_tables`.
> 
> Missing: re-add the same check as in [c49fead](https://github.com/fractal-analytics-platform/fractal-tasks-core/commit/c49feadb7580fee3349dfec3d74c98d37a04871d) (that is: factor it out into another module).

Adding this additional check in `apply_registration_to_ROI_tables` is not really relevant, since
1. The table has already been checked as part of `calculate_registration_image_based`, and one should likely re-use the same `roi_table` argument in the next task.
2. If, by mistake, a user sets the `roi_table` argument of `apply_registration_to_ROI_tables(..)` to a non-ROI table (e.g. something like `roi_table="measurements_DAPI"`), then the task will already fail upon calling `are_ROI_table_columns_valid`.

TL;DR: this issue should now be covered by #613
